### PR TITLE
Remove some rowkey functions

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

Hi, I'm studying OB's code and find some functions marked "TODO: remove", so I tried to remove two of them.
This PR is mainly to remove the one parameter version of `ObObj::check_collation_free_and_compare` and `ObRowKey::compare_prefix`.
There are some related functions (such as `ObRowKey::compare`) that are more complex to remove. I would like to continue to fix them later if possible.

### CODE REVIEW

**Comment by aim-for-better** [comment]
👍

**Review by jackysp — APPROVED** [review]
LGTM

**Comment by footka** [comment]
👍

**Comment by 7yyo** [comment]
LGTM

File: deps/oblib/src/common/rowkey/ob_rowkey.h [hunk 1]
```diff
@@ -189,7 +189,7 @@ class ObRowkey {
     if (obj_ptr_ == rhs.obj_ptr_) {
       cmp = static_cast<int32_t>(obj_cnt_ - rhs.obj_cnt_);
     } else {
-      cmp = compare_prefix(rhs);
+      compare_prefix(rhs, cmp);
```
Comment (simonjoylet) [inline deps/oblib/src/common/rowkey/ob_rowkey.h:192]: need check return code here. if you ignore the return code on purpose, you need write a comment to explain it or add a TODO task.

Comment (disksing) [reply deps/oblib/src/common/rowkey/ob_rowkey.h:194]: Hi @simonjoylet I have added some comment.

**Review by simonjoylet — APPROVED** [review]

**Review by LINxiansheng — APPROVED** [review]

**Comment by LINxiansheng** [comment]
Merged ,thanks for your contributions.

**Pull request merged**